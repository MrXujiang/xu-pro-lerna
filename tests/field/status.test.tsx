import { render } from 'enzyme';
import React from 'react';
import Field from '@ant-design/pro-field';

describe('Field Status', () => {
  const statusList = [
    'Success',
    'Error',
    'Processing',
    'Default',
    'Warning',
    'success',
    'error',
    'processing',
    'default',
    'warning',
  ];
  statusList.forEach((status) => {
    it(`ðŸ¥© ${status} render`, async () => {
      const html = render(
        <Field
          text="open"
          valueEnum={{
            open: {
              text: 'æœªè§£å†³',
              status,
            },
          }}
          mode="read"
        />,
      );
      expect(html).toMatchSnapshot();
    });
  });

  it(`ðŸ¥© red color render`, async () => {
    const html = render(
      <Field
        text="open"
        valueEnum={{
          open: {
            text: 'æœªè§£å†³',
            color: 'red',
          },
        }}
        mode="read"
      />,
    );
    expect(html).toMatchSnapshot();
  });
});
