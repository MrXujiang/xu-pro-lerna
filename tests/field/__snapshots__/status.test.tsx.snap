// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Status ðŸ¥© Default render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-default"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© Error render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-error"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© Processing render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-processing"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© Success render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-success"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© Warning render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-warning"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© default render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-default"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© error render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-error"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© processing render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-processing"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© red color render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-red"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© success render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-success"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;

exports[`Field Status ðŸ¥© warning render 1`] = `
<span
  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
>
  <span
    class="ant-badge-status-dot ant-badge-status-warning"
  />
  <span
    class="ant-badge-status-text"
  >
    æœªè§£å†³
  </span>
</span>
`;
